lib/gui/modules/drive-scanner.js

```diff
@@ -16,86 +16,140 @@
 
 'use strict';
 
-const Rx = require('rx');
 const _ = require('lodash');
 const EventEmitter = require('events').EventEmitter;
 const drivelist = require('drivelist');
+const debug = require('debug')('etcher:drive-scanner');
 const settings = require('../models/settings');
 
-const DRIVE_SCANNER_INTERVAL_MS = 2000;
-const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 0;
-const emitter = new EventEmitter();
+/**
+ * @summary Time to wait in between scan calls
+ * @type {Number}
+ * @constant
+ */
+const DRIVE_SCANNER_DELAY_MS = 2000;
+
+/**
+ * @summary Time to delay the first scan (in milliseconds)
+ * @description This prevents the drive scans from kicking
+ * off while Etcher is still loading, which would
+ * considerably delay the time until the main window shows
+ * @type {Number}
+ * @constant
+ */
+const DRIVE_SCANNER_FIRST_SCAN_DELAY_MS = 125;
 
-/* eslint-disable lodash/prefer-lodash-method */
+/**
+ * @summary DriveScanner class
+ * @class
+ * @extends {EventEmitter}
+ * @example
+ * var scanner = new DriveScanner()
+ *   .on('error', (error) => { ... })
+ *   .on('drives', (drives) => { ... })
+ */
+class DriveScanner extends EventEmitter {
 
-const availableDrives = Rx.Observable.timer(
-  DRIVE_SCANNER_FIRST_SCAN_DELAY_MS,
-  DRIVE_SCANNER_INTERVAL_MS
-)
+  /**
+   * @summary DriveScanner constructor
+   * @param {Object} [options] - EventEmitter options
+   * @example
+   * var scanner = new DriveScanner()
+   *   .on('error', (error) => { ... })
+   *   .on('drives', (drives) => { ... })
+   */
+  constructor(options) {
+    super(options);
+    this.isRunning = false;
+    this.timer = null;
+    this.scanCount = 0;
+    this.errorCount = 0;
+    this.run = this.run.bind(this);
+  }
 
-/* eslint-enable lodash/prefer-lodash-method */
+  /**
+   * @summary Start the scanner
+   * @public
+   * @example
+   * scanner.start()
+   */
+  start() {
+    debug('start');
+    this.isRunning = true;
+    this.timer = setTimeout(this.run, DRIVE_SCANNER_FIRST_SCAN_DELAY_MS);
+  }
 
-  .flatMap(() => {
-    return Rx.Observable.fromNodeCallback(drivelist.list)();
-  })
+  /**
+   * @summary Kick off a scan & schedule the next run
+   * @private
+   * @example
+   * scanner.run()
+   */
+  run() {
 
-  .map((drives) => {
-    if (settings.get('unsafeMode')) {
-      return drives;
+    if (!this.isRunning) {
+      return;
     }
 
-    return _.reject(drives, {
-      system: true
+    this.scanCount += 1;
+    DriveScanner.scan((error, drives) => {
+      debug('scan', error || drives);
+      if (error) {
+        this.errorCount += 1;
+        this.emit('error', error);
+      } else {
+        this.emit('drives', drives);
+      }
+      if (this.isRunning) {
```

Comment:
Is it necessary to check this here? If `isRunning` is false, then the next `this.run` call will already `return`, right?